### PR TITLE
Add SmileBASIC

### PR DESCRIPTION
**Pull request by TheV360**
Adds SmileBASIC.

*Automatically generated by PR builder*

### CODE REVIEW

File: games/smilebasic/game.dat [hunk 1]
```diff
@@ -0,0 +1,10 @@
+title = "SmileBASIC"
+description = "Program with BASIC on your 3DS! Made by SmileBoom Co.Ltd."
```
Comment (CodingKoopa) [inline games/smilebasic/game.dat]: This should be longer, and more objective, e.g. "SmileBASIC is a game where you can program...3DS. It was developer and published by..."

File: games/smilebasic/game.dat [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+needs_system_files = true
+needs_shared_font = true
+game_type = "eshop"
+
+[[releases]]
+title = "000400000016DE00"
+region = "jpn"
```
Comment (CodingKoopa) [inline games/smilebasic/game.dat]: I'm not sure, but I think this game would classify as `all`?

File: games/smilebasic/game.dat [hunk 3]
```diff
@@ -0,0 +1,10 @@
+title = "SmileBASIC"
+description = "SmileBASIC is a software development application for the Nintendo 3DS, developed and published by SmileBoom Co.Ltd.. Like its predecessor Petit Computer, SmileBASIC is built around a custom dialect of BASIC called SmileBASIC. With SmileBASIC, users can write programs that use the 3DS's features and then upload the programs (or folders containing programs) to an online repository, where they can be downloaded by entering a short code. Downloaded programs can also be edited and re-uploaded. Programs can utilize the 3DS's touchscreen, controls, speakers, gyroscopes, and 3D screen."
+needs_system_files = true
+needs_shared_font = true
+game_type = "eshop"
+
+[[releases]]
+title = "000400000016DE00"
+region = "all"
```
Comment (Leo626) [inline games/smilebasic/game.dat]: This is not correct. This is for `usa`.

**Comment by TheV360** [comment]
All issues are fixed. [The wiki entry has been created.](https://github.com/citra-emu/citra-games-wiki/wiki/SmileBASIC)

**Comment by Leo626** [comment]
Just a small nit, order the releases as `usa`, `eur` and `jpn`.

**Pull request merged**